File: 145/Solution.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    void helper(TreeNode* root, vector<int>& save){
        if(root){
            if(root->left){
                helper(root->left,save);
            }
            if(root->right){
                helper(root->right,save);
            }
            save.push_back(root->val);
        }
    }
    vector<int> postorderTraversal(TreeNode* root) {
        vector<int> save;
        helper(root,save);
        return save;
    }
};

File: 145/Solution.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public static void helper(TreeNode root, ArrayList<Integer> save){
        if(root!=null){
        if(root.left!=null){
            helper(root.left,save);
        }
        if(root.right!=null){
            helper(root.right,save);
        }
        save.add(root.val);
        }
    }

    public List<Integer> postorderTraversal(TreeNode root) {
        ArrayList<Integer> save = new ArrayList<>();
        helper(root,save);
        return save;
    }
}